src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -342,7 +346,7 @@ private static string GetPreferredUsernameFromIdToken(bool isAdfsAuthority, IdTo
                 return null;
             }
 
-            MsalAccessTokenCacheItem msalAccessTokenCacheItem = GetSingleResult(requestParams, finalList);
+            MsalAccessTokenCacheItem msalAccessTokenCacheItem = (MsalAccessTokenCacheItem)GetSingleResult(requestParams, finalList);
```

Comment:
I think it might be better to create two seperate methods that call into GetSingleResult but with the correct cache item type to avoid having to cast each time the method is used. 
Example: MsalAccessTokenCacheItem GetSingleAccessTokenResult(IReadOnlyList<MsalAccessTokenCacheItem> filteredItems....)

Reply:
Added